modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -0,0 +1,549 @@
+
+            ringByteBuffer.init(0);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected void body() throws InterruptedException, IgniteInterruptedCheckedException {
+            try {
+                while (!isCancelled()) {
+                    blockingSectionBegin();
+
+                    try {
+                        synchronized (this) {
+                            while (readyForFlushSize.get() < flushBatchSize && !isCancelled())
+                                wait();
+                        }
+                    }
+                    finally {
+                        blockingSectionEnd();
+                    }
+
+                    flushBuffer();
+                }
+            }
+            finally {
+                fileWriter = null;
+
+                ringByteBuffer.close();
+
+                // Make sure that all producers released their buffers to safe deallocate memory.
+                flushBuffer();
+
+                ringByteBuffer.free();
+
+                U.closeQuiet(fileIo);
+
+                cachedStrings.clear();
+
+                log.info("Performance statistics writer stopped.");
+            }
+        }
+
+        /** @return Write segment.*/
+        SegmentedRingByteBuffer.WriteSegment writeSegment(int size) {
+            SegmentedRingByteBuffer.WriteSegment seg = ringByteBuffer.offer(size);
+
+            if (seg != null) {
+                int readySize = readyForFlushSize.addAndGet(size);
+
+                if (readySize >= DFLT_FLUSH_SIZE) {
+                    synchronized (this) {
+                        // Required to start writing data to the file.
+                        notify();
+                    }
+                }
+            }
+
+            return seg;
+        }
+
+        /** Flushes to disk available bytes from the ring buffer. */
+        private void flushBuffer() {
+            List<SegmentedRingByteBuffer.ReadSegment> segs = ringByteBuffer.poll();
+
+            if (segs == null)
+                return;
+
+            try {
+                for (int i = 0; i < segs.size(); i++) {
+                    updateHeartbeat();
+
+                    SegmentedRingByteBuffer.ReadSegment seg = segs.get(i);
+
+                    try {
+                        readyForFlushSize.addAndGet(-seg.buffer().remaining());
+
+                        fileIo.writeFully(seg.buffer());
+                    }
+                    finally {
+                        seg.release();
+                    }
+                }
+
+                fileIo.force();
+            } catch (IOException e) {
+                log.error("Unable to write to file. Performance statistics collecting will be stopped.", e);
+
+                fileWriter.shutdown();
+
+                stopStatistics();
+            }
+        }
+
+        /** Shutted down the worker. */
+        private void shutdown() {
```

Comment:
We should override or use existing `cancel` method to stop this worker.

Reply:
Yes, existing `cancel` method can be used